pex/requirements.py

```diff
@@ -377,15 +377,64 @@ def _try_parse_project_name_from_path(path):
     return None
 
 
-def _try_parse_pip_local_project_marker(path):
-    # type: (str) -> Tuple[str, Optional[Marker]]
+def _try_parse_pip_local_formats(
+    path,  # type: str
+    basepath=None,  # type: Optional[str]
+):
+    # type: (...) -> Tuple[Optional[str], Optional[Marker]]
```

Comment:
Worth a dataclass return type?

Reply:
Pex still needs to be Python 2.7 compatible - that's why you see type comments and not type annotations.

Reply:
... and since this is a private datatype, ginning up a namedtuple or similar does not seem worth it.